packages/block-library/src/navigation-link/index.php

```diff
@@ -127,10 +127,10 @@ function render_block_core_navigation_link( $attributes, $content, $block ) {
 	$is_post_type           = isset( $attributes['kind'] ) && 'post-type' === $attributes['kind'];
 	$is_post_type           = $is_post_type || isset( $attributes['type'] ) && ( 'post' === $attributes['type'] || 'page' === $attributes['type'] );
 
-	// Don't render the block's subtree if it is a draft.
+	// Don't render the block's subtree if it is a draft or if the ID does not exist.
 	if ( $is_post_type && $navigation_link_has_id ) {
 		$post = get_post( $attributes['id'] );
-		if ( 'publish' !== $post->post_status ) {
+		if ( ! isset( $post ) || 'publish' !== $post->post_status ) {
```

Comment:
```suggestion
		if ( ! ( $post instanceof WP_Post ) || 'publish' !== $post->post_status ) {
```

Reply:
```suggestion
		if ( 'publish' !== get_post_status( $attributes['id'] ) ) {
```

Reply:
Thank you for the review @spacedmonkey. Can you include a note about which of the two proposed changes is the preferred solution, and why? It would help me out.

Reply:
I am leaning towards `get_post_status` (and removing `$post = get_post( $attributes['id'] );` 
but I do not know which is most performant. Untested.

Reply:
Technically, `isset` will work, so it's not wrong... According to the PHP docs,
> Determine if a variable is considered set, this means if a variable is declared and is different than **null**.

but on the other hand, `isset()` is not required here.
Since `get_post` returns either `WP_Post` or `null`, `! isset( $post )` can just be written `! $post`. Or `null === $post`. But that's just a matter of personal preference and there's nothing wrong with it 🤷